### PR TITLE
[CIVIS-9315] Fix apt-get and update AWS CLI

### PR DESCRIPTION
**Pull request by mheilman**
This fixes `apt-get` by updating the base image.

Also:
* Updates the AWS CLI to v2.
* Makes `generate-requirements-full.sh` run in a docker container to ensure the environment matches.
* Updates the code build config to use a FIPS URL (see, e.g., https://github.com/civisanalytics/datascience-r/pull/52) and to run builds for branches.
* Updates the major version to 8 since the AWS CLI major version changed.
* Addresses https://github.com/civisanalytics/datascience-python/security/dependabot/10

[Here's](https://platform.civisanalytics.com/spa/#/scripts/containers/282561515) a test job in Platform.

---

- [x] (For Civis employees only) Reference to a relevant ticket in the pull request title
- [x] Changelog entry added to `CHANGELOG.md` at the repo's root level
- [x] Description of change in the pull request description
- [x] If applicable, unit tests have been added and/or updated
- [x] The CircleCI builds have all passed

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -1,6 +1,13 @@
-FROM python:3.12.4-slim AS production
+ARG PLATFORM=linux/x86_64
+ARG BASE_IMAGE=python:3.12.5-slim
```
Comment (mheilman) [inline Dockerfile:2]: Just moving from `python:3.12.4-slim` to `python:3.12.5-slim` fixed the issue with running `apt-get` in FIPS-mode.  Apparently, `python:3.12.4-slim` was last built before the issue was fixed.

File: Dockerfile [hunk 2]
```diff
@@ -19,7 +26,9 @@ RUN DEBIAN_FRONTEND=noninteractive apt-get update -y --no-install-recommends &&
   libxrender1 \
   wget \
   ca-certificates \
-  curl && \
+  curl \
+  mandoc # For AWS CLI help \
+  unzip && \
```
Comment (mheilman) [inline Dockerfile:31]: unzip is needed for the aws CLI, but it seems generally useful and isn't a big dependency.

File: buildspec/push.yaml [hunk 3]
```diff
@@ -0,0 +1,19 @@
+version: 0.2
```
Comment (mheilman) [inline buildspec/push.yaml:1]: I added this to make things easier to test.

**Comment by mheilman** [comment]
Oh, there's a dependabot alert: https://github.com/civisanalytics/datascience-python/security/dependabot/10

I'll try to fix that

**Comment by mheilman** [comment]
With regards to the dependabot alert, `requirements-full.txt` says that `certifi` is just pulled from `requests`, but we're on the latest version of `requests`, and it [doesn't specify](https://github.com/psf/requests/blob/a3ce6f007597f14029e6b6f54676c34196aa050e/setup.py#L40) a max version for `certifi`, so I'm not sure why the latest version isn't used.  I'm going to leave this for later.

**Review by jacksonlee-civis — APPROVED** [review]
LGTM

**Pull request merged**